### PR TITLE
Broadcastbutton

### PR DESCRIPTION
**Pull request by realswat1**
This PR is for the broadcast button . This is to help get the stream key to the ui by hitting the broadcast endpoint . I had to make some changes the usercontext file , login and main file to allow that the usercontext is available to all components. The broadcast endpoint is protected by the jwt . The test place attached demonstrates how this works. 
https://www.loom.com/share/543ccec4af4c4935997032488fe43004

### CODE REVIEW

File: exposee-ui/src/components/Login/Login.jsx [hunk 1]
```diff
@@ -30,10 +30,15 @@ const Login = () => {
...
-
         // Update the user context
         updateUser(loggedInUser);
         localStorage.setItem('access_token', access_token);
+        const userWithToken = {
+          ...loggedInUser,
+          access_token: access_token,
+        };
+        updateUser(userWithToken);
+        console.log('hello', userWithToken )
```
Comment (alfonsocheng) [inline exposee-ui/src/components/Login/Login.jsx]: probably would remove this before merging in

File: exposee-ui/src/components/Profilepage/Profile.jsx [hunk 2]
```diff
@@ -1,46 +1,78 @@
...
-            const response = await fetch ('/broadcast');
-            console.log(Response.data);
-        }catch (error){
-            console.error('Error going live: ', error);
-        }
-    }; console.log(user.profilePicture)
-    if (!user){
-        return <div>Loading...</div>;
+  }, [user.id, setUser]);
+console.log(user.access_token);
```
Comment (alfonsocheng) [inline exposee-ui/src/components/Profilepage/Profile.jsx]: probably also do not want to console log the user access token

File: exposee-ui/src/components/Profilepage/Profile.jsx [hunk 3]
```diff
@@ -1,46 +1,78 @@
...
+      } else {
+        console.log("Failed to start the live stream");
+      }
+    } catch (error) {
+      console.error('Error going live:', error);
     }
-    return (
+  };
+
+  if (isLoading) {
```
Comment (alfonsocheng) [inline exposee-ui/src/components/Profilepage/Profile.jsx:56]: lol it seems you already did a loading spinner for the profile page??

File: exposee-ui/src/main.jsx [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
 import ReactDOM from 'react-dom/client'
 import App from './App' 
+import { UserProvider } from './UserContext'
 import './index.css'
 
 ReactDOM.createRoot(document.getElementById('root')).render(
   <React.StrictMode>
-    <App />
+    <UserProvider>
+    <App />
```
Comment (alfonsocheng) [inline exposee-ui/src/main.jsx:10]: I'm surprised eslint didn't indent the App element

**Review by alfonsocheng — APPROVED** [review]

File: exposee-ui/src/components/Profilepage/Profile.jsx (hunk 3 above)
Comment (realswat1) [reply exposee-ui/src/components/Profilepage/Profile.jsx:56]: i thought u wanted it to be in the video
.. I told me if i am feeling fancy i can do it on the video too

File: exposee-ui/src/main.jsx (hunk 4 above)
Comment (realswat1) [reply exposee-ui/src/main.jsx:10]: i didn't run it because it messed up the first time

**Pull request merged**